save-backend/src/main/kotlin/com/saveourtool/save/backend/service/TestExecutionService.kt

```diff
@@ -224,34 +220,31 @@ class TestExecutionService(private val testExecutionRepository: TestExecutionRep
                         log.error("Test execution $testExecDto with id=$testExecutionId for execution id=$executionId cannot be updated because its status is not RUNNING")
                     })
         }
-        val lock = locks.computeIfAbsent(executionId) { Any() }
-        synchronized(lock) {
-            transactionTemplate.execute {
-                val execution = executionRepository.findById(executionId).get()
-                execution.apply {
-                    log.debug {
-                        "Updating counters in execution id=$executionId: running=$runningTests-${counters.total()}, " +
-                                "passed=$passedTests+${counters.passed}, failed=$failedTests+${counters.failed}, skipped=$skippedTests+${counters.skipped}"
-                    }
-                    runningTests -= counters.total()
-                    passedTests += counters.passed
-                    failedTests += counters.failed
-                    skippedTests += counters.skipped
+        transactionTemplate.execute {
+            val execution = executionRepository.findWithLockingById(executionId).orElse(null).orNotFound()
```

Comment:
what's about removing all these calculation on Execution and move it to Runtime and calculate them by request using `TestExecutionDto`?
something like:
```kotlin
val counters = Counters()
testExecutionRepository.findByExecutionId(executionId)
.forEach {
    when (it.status) {
        TestResultStatus.PASSED -> counters.passed++
        TestResultStatus.FAILED -> counters.failed++
        else -> counters.skipped++
    }

    with(counters) {
        unmatchedChecks += it.unmatched.orZeroIfNotApplicable()
        matchedChecks += it.matched.orZeroIfNotApplicable()
        expectedChecks += it.expected.orZeroIfNotApplicable()
        unexpectedChecks += it.unexpected.orZeroIfNotApplicable()
    }
}
```

and then return it when it's required?
Execution can be updated finally when status is changed to terminated state (or during archiving)

Reply:
raised: https://github.com/saveourtool/save-cloud/issues/1480